File: Data Structures & Algorithms/three-integer-sum/submission-5.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number[][]}
     */
    threeSum(nums) {

        nums.sort((a,b) => a - b);

        let result = [];

        for(let i = 0; i < nums.length - 2; i++) {
            if(i > 0 && nums[i] === nums[i-1]) {
                continue;
            }
            for(let j = i + 1; j < nums.length - 1; j++) {
                if(j > i + 1 && nums[j] === nums[j-1]) {
                    continue;
                }
                for(let k = j + 1; k < nums.length; k++) {
                    if(k > j + 1 && nums[k] === nums[k-1]) {
                        continue;
                    }
                    if(nums[i] + nums[j] + nums[k] === 0) {
                        result.push([nums[i], nums[j], nums[k]])
                    }
                }
            }
        }

        return result;

    }
}

File: Data Structures & Algorithms/remove-element/submission-5.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        k = 0

        while k < len(nums) and nums[k] != val:
            k += 1

        if k == len(nums):
            return k

        for i in range(k + 1, len(nums)):
            if nums[i] != val:
                nums[k] = nums[i]
                k += 1
            # while k < len(nums) and nums[k] != val:
            #     k += 1
                 
        return k